textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcher.java

```diff
@@ -34,22 +52,58 @@ public void open() throws DataFlowException {
 
     @Override
     public ITuple getNextTuple() throws DataFlowException {
-
-        try {
+		try {
             ITuple sourceTuple = sourceOperator.getNextTuple();
             if(sourceTuple == null){
                 return null;
             }
-            if (predicate.satisfy(sourceTuple)) {
-                return sourceTuple;
-            } else {
-                return getNextTuple();
+            
+            RegexPredicate regexPredicate = (RegexPredicate)predicate; 
+            
+            spans = regexPredicate.statisfySpan(sourceTuple);
+            
+            if (spans.size() != 0) { // a list of matches found
```

Comment:
According to the implementation of satisfySpan(computeMatches),  "spans" won't be NULL. But adding this will make it safer.